### ISSUE TITLE
Getting "maximum recursion depth exceeded" because of ScheduledObserver

### ISSUE THREAD

**Issue by MichaelSchneeberger** [ReactiveX/RxPY#184]
The following code ...

```
from rx import AnonymousObserver
from rx.concurrency import current_thread_scheduler
from rx.core.scheduledobserver import ScheduledObserver

obs1 = AnonymousObserver()
scheduled_obs1 = ScheduledObserver(scheduler=current_thread_scheduler, observer=obs1)

for i in range(1100):
    scheduled_obs1.on_next(i)

scheduled_obs1.on_completed()
scheduled_obs1.ensure_active()
scheduled_obs1.dispose()
```

... produces the following traceback ...

  _[...]
  File "C:\Users\chmisch7\AppData\Local\Continuum\Anaconda3\envs\hilbert\lib\site-packages\rx\disposables\singleassignmentdisposable.py", line 55, in dispose
    old.dispose()
  [Previous line repeated **992** more times]
  File "C:\Users\chmisch7\AppData\Local\Continuum\Anaconda3\envs\hilbert\lib\site-packages\rx\disposables\singleassignmentdisposable.py", line 48, in dispose
    with self.lock:
RecursionError: **maximum recursion depth exceeded** while calling a Python object_

It is due to recursive scheduling of the function 'ScheduledObserver.run' (line 67), which builds a stacked disposable.

```
def run(self, recurse, state):
    [...]
    return self.scheduler.schedule(self.run)
```

Calling the dispose function on that stacked disposable will recursively call all dispose functions and lead to an exceeded maximum recursion depth.

**Comment by MichaelSchneeberger** [comment]
One way to avoid this is to replace ...

```
    return self.scheduler.schedule(self.run)
```

... by ...

```
    from . import Disposable
    disposable = self.scheduler.schedule(self.run)
    return Disposable.create(lambda: self.scheduler.schedule(lambda r, s: disposable.dispose()))
```

**Comment by dbrattli** [comment]
Thanks for reporting this. Looks like a bug. I think I will replace it with a ScheduledDisposable which looks a bit better than the lambda, and add a unit-test to make sure this doesn't happen again..

```python
    disposable = self.scheduler.schedule(self.run)
    return ScheduledDisposable(scheduler, disposable)
```

**Comment by dbrattli** [comment]
This should be fixed in version 1.6.1
